src/value.c

```diff
@@ -259,6 +273,41 @@ const char *SIType_ToString(SIType t) {
 	}
 }
 
+void SIType_ToMultipleTypeString(SIType t, char **buf, size_t *bufferLen, size_t *bytesWritten) {
+	// Worst case: Len(SIType names) + 19*Len(", ") + Len("Or") = 177 + 38 + 2 = 217
+	ASSERT((*bufferLen - *bytesWritten) >= 256);
+
+	SIType currentType;
+	SIType remainingTypes = t;
+	
+	uint typesDetected = 0;
+	// Iterate over the possible SITypes
+	for(currentType = 1; currentType < SI_ALL; currentType = currentType << 1) {
+		if(t & currentType) {
+			remainingTypes &= (~currentType);
+			if(typesDetected > 0) {
+				if(remainingTypes) {
+					*bytesWritten += snprintf(*buf + *bytesWritten, *bufferLen, ", ");
+				} else {
+					// There is no more types, then concatenate "or" before the SIType name
+					if(typesDetected == 1) {
+						// If there are two items, the last comma should be omitted.
+						*bytesWritten += snprintf(*buf + *bytesWritten, *bufferLen, " or ");
+					} else {
+						// If there are more than two, the last comma should be present
+						*bytesWritten += snprintf(*buf + *bytesWritten, *bufferLen, ", or ");
+					}
+				}
+			}
+			*bytesWritten += snprintf(*buf + *bytesWritten, *bufferLen, "%s", SIType_ToString(currentType));
+			typesDetected++;
+		}
+		if(!remainingTypes) {
+			break;
+		}
```

Comment:
using `popcount` this condition should be removed.

Reply:
Ok. Done. Thanks.